xml/depl_maintenance.xml

```diff
@@ -508,6 +508,275 @@
+       (preferably the one with the <filename>/etc/pg2mysql/databases.yaml</filename> file).
+      </para>
+     </step>
+     <step>
+      <para>
+       To determine whether the &postgres; databases contain data that cannot
+       be migrated to &mariadb;, run <command>psql2mysql</command> with the
+      <literal>precheck</literal> option:</para>
+<screen>&prompt.user;psql2mysql \
+--source postgresql://keystone:vZn3nfxXzv97@192.168.243.87/keystone \
+--target mysql+pymysql://keystone:vZn3nfxXzv97@192.168.243.88/keystone?charset=utf8 \
+precheck</screen>
+      <para>
+       To run precheck operation on all databases in a single operation, use the
+       <literal>--batch</literal> option and the
+       <filename>/etc/pg2mysql/databases.yaml</filename> file as follows:
+      </para>
+<screen>&prompt.user;psql2mysql --batch /etc/pg2mysql/databases.yaml precheck</screen>
+      <para>
+       If the precheck indicates that there is data that cannot be imported
+       into &mariadb;, modify the offending data manually to fix
+       the problems. The example below shows what an output containing issues may
+       look like:
+      </para>
+ <screen>&prompt.user;psql2mysql --source postgresql://cinder:idRll2gJPodv@192.168.243.86/cinder precheck
+Table 'volumes' contains 4 Byte UTF8 characters which are incompatible with the 'utf8' encoding used by MariaDB
+The following rows are affected:
++-----------------------------------------+-----------------+-------+
+|               Primary Key               | Affected Column | Value |
++-----------------------------------------+-----------------+-------+
+| id=5c6b0274-d18d-4153-9fda-ef3d74ab4500 |   display_name  |   💫   |
++-----------------------------------------+-----------------+-------+
+Error during prechecks. 4 Byte UTF8 characters found in the source database.</screen>
+     </step>
+     <step>
+      <para>
+       Stop chef-client services on the nodes, to prevent regular runs of
+       chef-client from starting database-related &ostack; services again. To
+       do this from the &admnode;, you can use the <command>knife ssh roles:dns-client systemctl
+       stop chef-client</command> command. Stop
+       all &ostack; services that make use of the database to prevent them from
+       writing new data during the migration.
+      </para>
+      <note>
+       <title>Testing Migration Procedure</title>
+       <para>
+	If you want to perform a dry run of the migration procedure, you can run the
+       <command>psql2mysql migrate</command> without stopping the
+       database-related &ostack; services. This way, if the test migration
+       fails due to errors that weren't caught by the precheck procedure, you
+       can fix them with &ostack; services still running, thus minimizing the
+       required downtime. When you perform the actual migration, the data in
+       the target databases will be replaced with the latest one in the source databases.
```

Comment:
Please mention also the option to use the `purge-tables` command. After such test migration, we advice users to purge the tables in target database before running the real migration, just to speed up the real migration. It's not necessary, data would be deleted anyway, but it will save some time.

See https://github.com/SUSE/psql2mysql/blob/master/README.md and search for purge-tables

Reply:
Fixed.